webpack/JobWizard/steps/ReviewDetails/index.js

```diff
@@ -0,0 +1,202 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Button,
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+} from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
+import { get } from 'foremanReact/redux/API';
+import { translate as __, sprintf } from 'foremanReact/common/I18n';
+import {
+  selectJobTemplates,
+  selectHosts,
+  selectHostCount,
+  selectTemplateInputs,
+  selectAdvancedTemplateInputs,
+} from '../../JobWizardSelectors';
+import {
+  HOSTS_API,
+  HOSTS_TO_PREVIEW_AMOUNT,
+  WIZARD_TITLES,
+} from '../../JobWizardConstants';
+import { buildHostQuery } from '../HostsAndInputs/buildHostQuery';
+import { WizardTitle } from '../form/WizardTitle';
+
+const ReviewDetails = ({
+  jobCategory,
+  jobTemplateID,
+  advancedValues,
+  scheduleValue,
+  templateValues,
+  selectedTargets,
+  hostsSearchQuery,
+}) => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(
+      get({
+        key: HOSTS_API,
+        url: '/api/hosts',
+        params: {
+          search: buildHostQuery(selectedTargets, hostsSearchQuery),
+          per_page: HOSTS_TO_PREVIEW_AMOUNT,
+        },
+      })
+    );
+  }, [dispatch, hostsSearchQuery, selectedTargets]);
+  const jobTemplates = useSelector(selectJobTemplates);
+  const templateInputs = useSelector(selectTemplateInputs);
+  const advancedTemplateInputs = useSelector(selectAdvancedTemplateInputs);
+  const jobTemplate = jobTemplates.find(
+    template => template.id === jobTemplateID
+  )?.name;
+
+  const hosts = useSelector(selectHosts);
+
+  const hostsCount = useSelector(selectHostCount);
+  const stringHosts = () => {
+    if (hosts.length === 0) {
+      return __('No Target Hosts');
+    }
+    if (hosts.length === 1 || hosts.length === 2) {
+      return hosts.join(', ');
+    }
+    return `${hosts[0]}, ${hosts[1]} ${sprintf(
+      __(', and %s more'),
+      hostsCount - 2
+    )}`;
+  };
+  const [isAdvancedShown, setIsAdvancedShown] = useState(false);
+  const detailsFirstHalf = [
+    { label: __('Job Category'), value: jobCategory },
+    { label: __('Job template'), value: jobTemplate },
+    { label: __('Target hosts'), value: stringHosts() },
```

Comment:
Can we use the same collapse functionality as `Advanced fields`
if the hosts lists will be long, it's pretty hard to see the affected hosts:
![Screenshot (39)](https://user-images.githubusercontent.com/26363699/138277608-e1844d3e-3ec0-4e64-a8e0-b46f0fac6a54.png)

Reply:
I think as a summary, user should not go back to the hosts step. We could rather reuse the hosts modal in the summary: https://github.com/theforeman/foreman_remote_execution/pull/611#discussion_r731007851

Reply:
I have no issue with it being a number, although currently it's a text with tooltip... but let's allow clicking on it and expand the section like in the advanced fields section where we also don't need to return to the advanced wisard's step.

If not expanding, we could show the same hosts modal as in the hosts wizard step.